File: Cloud Storage System/Level 1/python/file_repository.py
# file_repository.py

from file import File

class FileRepository:
    def __init__(self):
        self._storage = {}  # path -> File

    def exists(self, path: str) -> bool:
        return path in self._storage

    def add(self, file: File) -> bool:
        if self.exists(file.path):
            return False
        self._storage[file.path] = file
        return True

    def get(self, path: str) -> File | None:
        return self._storage.get(path)

    def delete(self, path: str) -> File | None:
        return self._storage.pop(path, None)

File: Cloud Storage System/Level 1/C#/file_repository.cs
// FileRepository.cs
using System.Collections.Generic;

public class FileRepository
{
    private readonly Dictionary<string, File> _storage = new();

    public bool Exists(string path)
    {
        return _storage.ContainsKey(path);
    }

    public bool Add(File file)
    {
        if (Exists(file.Path)) return false;
        _storage[file.Path] = file;
        return true;
    }

    public File? Get(string path)
    {
        return _storage.TryGetValue(path, out var file) ? file : null;
    }

    public File? Delete(string path)
    {
        if (!_storage.TryGetValue(path, out var file)) return null;
        _storage.Remove(path);
        return file;
    }
}
